dask/array/core.py

```diff
@@ -880,15 +879,24 @@ def __new__(cls, dask, name, chunks, dtype=None, meta=None, shape=None):
             raise TypeError("You must not specify both meta and dtype")
         if dtype is None and meta is None:
             raise ValueError("You must specify the meta or dtype of the array")
-
-        self._chunks = normalize_chunks(chunks, shape, dtype=dtype or meta.dtype)
+        if meta is not None:
+            meta = meta_from_array(meta)
+
+        if (isinstance(chunks, str) or
+                isinstance(chunks, tuple) and
+                chunks and
```

Comment:
Isn't `isinstance(chunks, str) or isinstance(chunks, tuple)` and `chunks` redundant? If `chunks` is either `str` or `tuple`, then it can't be `None` or `False`.

Reply:
I think this line can be removed for code clarity.

Reply:
But it can be `()`, which is falsey :)

Reply:
Hmm, I didn't know about that, thanks for clarifying!